contracts/cw1-whitelist-ng/src/multitest/contract.rs

```diff
@@ -0,0 +1,239 @@
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &Cw1ExecMsg::Execute { msgs },
+            self.send_funds,
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct WhitelistExecutor<'a, App> {
+    addr: Addr,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+}
+
+impl<'a, App> WhitelistExecutor<'a, App> {
+    pub fn new(addr: Addr, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            addr,
+            app,
+            sender,
+            send_funds,
+        }
+    }
+
+    pub fn freeze<C>(self) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + Serialize + 'static,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &WhitelistExecMsg::Freeze {},
+            self.send_funds,
+        )
+    }
+
+    pub fn update_admins<C>(self, admins: Vec<String>) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + 'static + Serialize,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &WhitelistExecMsg::UpdateAdmins { admins },
+            self.send_funds,
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct Instantiator<'a, App> {
+    code_id: u64,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+    label: String,
+    admin: Option<String>,
+}
+
+impl<'a, App> Instantiator<'a, App> {
+    pub fn new(code_id: u64, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            code_id,
+            app,
+            sender,
+            send_funds,
+            label: "Cw1Whitelist".to_owned(),
+            admin: None,
+        }
+    }
+
+    pub fn with_label(mut self, label: &str) -> Self {
+        self.label = label.to_owned();
+        self
+    }
+
+    pub fn with_admin(mut self, admin: &str) -> Self {
+        self.admin = Some(admin.to_owned());
+        self
+    }
+
+    pub fn with<C>(self, admins: Vec<String>, mutable: bool) -> AnyResult<Cw1WhitelistProxy>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + 'static,
+        App: Executor<C>,
+    {
+        let addr = self.app.instantiate_contract(
+            self.code_id,
+            self.sender,
+            &InstantiateMsg { admins, mutable },
+            self.send_funds,
+            self.label,
+            self.admin,
+        )?;
+
+        Ok(Cw1WhitelistProxy(addr))
+    }
+}
+
+// Proxy for direct execution in multitest.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Cw1WhitelistProxy(Addr);
+
+impl Cw1WhitelistProxy {
+    pub fn addr(&self) -> Addr {
+        self.0.clone()
+    }
+
+    pub fn instantiate<'a, App>(
```

Comment:
Hmmm... so I would really call it.

`let contract = Cw1WhitelistProxy::instantiate(...).with(...).unwrap();`

That makes more sense, but still... I could imagine a better name.
Not going to block merge with it, but happy for more naming feedback before this name gets ported to tons of auto-generated code.